sycl/doc/extensions/proposed/sycl_ext_oneapi_clock.asciidoc

```diff
@@ -0,0 +1,132 @@
+references below to the "core SYCL specification" or to section numbers in the
+SYCL specification refer to that revision.
+
+== Status
+
+This is a proposed extension specification, intended to gather community
+feedback.  Interfaces defined in this specification may not be implemented yet
+or may be in a preliminary state.  The specification itself may also change in
+incompatible ways before it is finalized.  *Shipping software products should
+not rely on APIs defined in this specification.*
+
+== Backend support status
+
+The APIs in this extension may be used only on a device that has
+`aspect::ext_oneapi_clock`.  The application must check that the device has
+this aspect before submitting a kernel using any of the APIs in this
+extension.  If the application fails to do this, the implementation throws
+a synchronous exception with the `errc::feature_not_supported` error code
+when the kernel is submitted to the queue.
+
+== Overview
+
+This extension defines a new free function `clock()`. It returns the sample
+value of a clock as seen by the shader processor. An idealized clock is an
+unbounded unsigned scalar integer tick count increasing monotonically over
+time. A clock’s rate of progress may vary within the lifetime of an invocation,
+may vary across different executions of the program, and may be affected by
+conditions beyond the control of the programmer. The sampled value read by this
+instruction consists of the least significant bits of the idealized clock’s tick
+count at the time the instruction was executed. In particular, an observer may
+see sampled values wrap around zero.
+
+== Specification
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification.  An implementation supporting this extension must predefine the
+macro `SYCL_EXT_ONEAPI_CLOCK` to one of the values defined in the table
+below.  Applications can test for the existence of this macro to determine if
+the implementation supports this feature, or applications can test the macro's
+value to determine which of the extension's features the implementation
+supports.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|Initial version of this extension.
+|===
+
+=== New device aspect
+
+This extension adds new device aspect:
+
+```c++
+namespace sycl::ext::oneapi {
+
+enum class aspect : /*unspecified*/ {
+  ext_oneapi_clock
+};
+
+} // namespace sycl::ext::oneapi
+```
+
+[width="100%",%header,cols="50%,50%"]
+|===
+|Aspect
+|Description
+
+|`ext_oneapi_clock`
+|Indicates that the device supports the `clock()` call.
+|===
+
+=== New free function
+
+```c++
+namespace sycl::ext::oneapi {
+
+uint64_t clock();
+
+} // namespace sycl::ext::oneapi
+```
+
+_Returns:_ the sample value of a clock as seen by the shader processor.
+
+_Throws:_ an exception with the errc::feature_not_supported error code if the
+device does not have aspect::ext_oneapi_clock.
```

Comment:
This function does not throw any exceptions, so I'd remove this paragraph.  The exception is thrown on the host from the kernel invocation function (e.g. `parallel_for`), but this is described above under "Backend support status".

Reply:
Fixed [571c693](https://github.com/intel/llvm/pull/19842/commits/571c6937ea4bc996f8eff440251425303c71af4e)